app/portainer/rbac/components/access-viewer/access-viewer.controller.js

```diff
@@ -101,6 +101,33 @@ export default class AccessViewerController {
     return this.findLowestRole(policyRoles);
   }
 
+  // for admin, returns all users
+  // for team leader, only return all his/her team member users
+  async teamMemberUsers(users, teamMemberships) {
+    if (this.isAdmin) {
+      return users;
+    }
+
+    const filteredUsers = [];
+    const userId = this.Authentication.getUserDetails().ID;
+    const leadingTeams = await this.UserService.userLeadingTeams(userId);
+
+    const isMember = (userId, teamId) => {
+      return !!_.find(teamMemberships, { UserId: userId, TeamId: teamId });
+    };
+
+    for (const user of users) {
```

Comment:
If possible, I think you should see if there is a more efficient solution using some combination of `filter`, `map` or `reduce` as I think nested for loops generally are inefficient/dangerous

Reply:
filter, map, and reduce are not more efficient, they do loop too.
This function has considered the balance between efficiency and readability. 
Welcome to raise an optimized implementation.

Reply:
Can't think of one so I will resolve this 👍🏻